### PR TITLE
Default folder for .list(), .readdir(), and .download()

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

**Comment by LuKks** [comment]
@mafintosh Ready!

File: index.js [hunk 1]
```diff
@@ -283,13 +284,15 @@ module.exports = class Hyperdrive extends EventEmitter {
...
   list (folder, { recursive = true } = {}) {
+    folder = folder || '/'
     if (folder.endsWith('/')) folder = folder.slice(0, -1)
     if (recursive === false) return shallowReadStream(this.files, folder, false)
     // '0' is binary +1 of /
     return folder ? this.entries({ gt: folder + '/', lt: folder + '0' }) : this.entries()
   }
 
   readdir (folder) {
+    folder = folder || '/'
```
Comment (mafintosh) [inline index.js:295]: if (!folder) folder = '/'

everywhere :)

Comment (mafintosh) [reply index.js:295]: or just folder = '/' in the args

File: index.js [hunk 2]
```diff
@@ -283,13 +284,15 @@ module.exports = class Hyperdrive extends EventEmitter {
 
   // atm always recursive, but we should add some depth thing to it
   list (folder, { recursive = true } = {}) {
+    folder = folder || '/'
```
Comment (mafintosh) [inline index.js:287]: needs `if (typeof folder === 'object') return this.list(undefined, folder)` also then

File: index.js [hunk 3]
```diff
@@ -265,6 +265,7 @@ module.exports = class Hyperdrive extends EventEmitter {
   }
 
   async download (folder, opts) {
+    folder = folder || '/'
```
Comment (mafintosh) [inline index.js:268]: same as below

File: index.js (hunk 1 above)
Comment (LuKks) [reply index.js]: Would it be better the condition to cover null values?

Comment (mafintosh) [reply index.js]: i'd just use the default arg, we rely on that elsewhere as well, so need need to be anymore defensive than that

File: index.js [hunk 4]
```diff
@@ -283,13 +284,16 @@ module.exports = class Hyperdrive extends EventEmitter {
 
   // atm always recursive, but we should add some depth thing to it
   list (folder, { recursive = true } = {}) {
+    if (typeof folder === 'object') return this.list(undefined, folder)
+    if (!folder) folder = '/'
```
Comment (mafintosh) [inline index.js:288]: move this to the default arg also

File: index.js [hunk 5]
```diff
@@ -283,13 +284,16 @@ module.exports = class Hyperdrive extends EventEmitter {
...
     if (folder.endsWith('/')) folder = folder.slice(0, -1)
     if (recursive === false) return shallowReadStream(this.files, folder, false)
     // '0' is binary +1 of /
     return folder ? this.entries({ gt: folder + '/', lt: folder + '0' }) : this.entries()
   }
 
   readdir (folder) {
+    if (!folder) folder = '/'
```
Comment (mafintosh) [inline index.js:296]: and this

File: index.js [hunk 6]
```diff
@@ -264,8 +264,7 @@ module.exports = class Hyperdrive extends EventEmitter {
     return this.files.createReadStream(opts)
   }
 
-  async download (folder, opts) {
-    if (!folder) folder = '/'
+  async download (folder = '/', opts) {
```
Comment (mafintosh) [inline index.js:267]: if (typeof folder === object ... here also

Comment (LuKks) [reply index.js:267]: Right, you want i.e. `download({ recursive: false })` to just work, nice!

**Pull request merged**